### PR TITLE
[3.15] Add `__class_getitem__` of classes

### PR DESCRIPTION
**Pull request by saJaeHyukc**
# I have made things!
I added `__class_getitem__` of classes  in DRF version 3.15.

The following are class items with `__class_getitem__` added.
- `fields.Field.__class_getitem__`
- `generics.GenericAPIView.__class_getitem__`
- `request.Request.__class_getitem__`
- `response.Response.__class_getitem__`

## Related issues

Upstream PR: https://github.com/encode/django-rest-framework/pull/8825

### CODE REVIEW

File: rest_framework-stubs/response.pyi [hunk 1]
```diff
@@ -22,6 +23,7 @@ class Response(SimpleTemplateResponse):
         exception: bool = ...,
         content_type: str | None = ...,
     ) -> None: ...
+    def __class_getitem__(cls, *args: Incomplete, **kwargs: Incomplete) -> Incomplete: ...
```
Comment (sobolevn) [inline rest_framework-stubs/response.pyi:26]: Please, provide correct types, `Incomplete` is not a good solution :(

Comment (saJaeHyukc) [reply rest_framework-stubs/response.pyi:26]: Thank you for letting me know the shortcomings! Is it okay to change only the 'response' part for that type, or should I revise the other parts as well?

> Please, provide correct types, `Incomplete` is not a good solution :(

**Comment by saJaeHyukc** [comment]
> 모든 장소를 이용해주세요 :) 예는 다음과 같습니다: https://github.com/search?q=repo%3Apython%2Ftypeshed%20__class_getitem__&type=code

Oh... I sincerely appreciate you providing an example!!
I will repost the PR once the necessary modifications are completed.

**Comment by saJaeHyukc** [comment]
> > All places, please :)
Here are examples: https://github.com/search?q=repo%3Apython%2Ftypeshed%20__class_getitem__&type=code
> 
> Oh... I sincerely appreciate you providing an example!! I will repost the PR once the necessary modifications are completed.

**Comment by saJaeHyukc** [comment]
With 'General Alias', only version 3.9 or later is compatible.

python docs
> https://docs.python.org/3/library/types.html#:~:text=int))%0ATrue-,New%20in%20version%203.9.,-Changed%20in%20version

<br>

Regarding your inquiry, is it better to coordinate the current 'Base Serializer' '__class_getitem__'?

>https://github.com/typeddjango/djangorestframework-stubs/blob/41773dd23a596b5f8dee10028b5cbbd1d451710d/rest_framework-stubs/serializers.pyi#L82

**Pull request closed without merging**